src/arrayMethodSort.js

```diff
@@ -4,8 +4,59 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
+      comparator = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    } else {
+      comparator = compareFunction;
+    }
+
+    function quickSort(arr, left, right) {
+      if (left < right) {
+        const pivotIndex = partition(arr, left, right);
+
+        quickSort(arr, left, pivotIndex - 1);
+
+        quickSort(arr, pivotIndex + 1, right);
+      }
+    }
```

Comment:
You've implemented the quicksort algorithm, which is a very efficient sorting algorithm for larger arrays. However, for smaller arrays, simpler algorithms like insertion sort can actually be faster due to their lower overhead. Consider using a hybrid approach: quicksort for larger arrays, and a simpler algorithm for smaller ones.